raft.go

```diff
@@ -656,14 +774,19 @@ func (r *Raft) runFollower() {
 			// Heartbeat failed! Transition to the candidate state
 			lastLeader := r.Leader()
 			r.setLeader("")
-			if len(r.peers) == 0 && !r.conf.EnableSingleNode {
+			if r.configurations.latestIndex == 0 {
 				if !didWarn {
-					r.logger.Printf("[WARN] raft: EnableSingleNode disabled, and no known peers. Aborting election.")
+					r.logger.Printf("[WARN] raft: no known peers, aborting election")
+					didWarn = true
+				}
+			} else if r.configurations.latestIndex == r.configurations.committedIndex &&
+				!hasVote(r.configurations.latest, r.localGUID) {
+				if !didWarn {
+					r.logger.Printf("[WARN] raft: not part of stable configuration, aborting election")
 					didWarn = true
 				}
 			} else {
```

Comment:
It feels like there's a hole here where a non-voter could try to elect itself if there's a config change in process (the indexes don't match). Do we need a blanket `hasVote` here in the final `else` clause?